pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -54,7 +54,7 @@ in
 , profilingDetail ? "exported-functions"
 # TODO enable shared libs for cross-compiling
 , enableSharedExecutables ? false
-, enableSharedLibraries ? !stdenv.hostPlatform.isStatic && (ghc.enableShared or false)
+, enableSharedLibraries ? !stdenv.hostPlatform.isStatic && (ghc.enableShared or false) && !stdenv.hostPlatform.useAndroidPrebuilt
```

Comment:
Otherwise
```
$ nix-build -A pkgsCross.aarch64-android-prebuilt.haskell.packages.ghc910.attoparsec
...
RPATH of binary /nix/store/g6ncgk2b04mv722hh2jgfbixyawbwabv-attoparsec-aarch64-unknown-linux-android-0.14.4/lib/ghc-9.10.1/lib/aarch64-android-ghc-9.10.1-inplace/libHSattoparsec-0.14.4-GkMCGqk5eKArKmd1r9IB7-ghc9.10.1.so contains a forbidden reference to /build/
```